### ISSUE TITLE
Regarding the issue of EfficientSAM of checkpoints

### ISSUE THREAD

**Issue by feivellau** [yformer/EfficientSAM#36]
Hello, I'm grateful for your research. 
I have a question about checkpoints. Is the change in weights of efficentsam compared to sam only the image encoder part.
Or can I get the correct result by processing the weights of the image encoder of efficentsam with the weights of the prompt encoder and mask decoder of the original sam?

**Comment by yformer** [comment]
@feivellau, thanks for your interest! We finetuned the entire model including encoder and decoder. I guess it may not work with efficientsam encoder and original sam decoder. Maybe you can give it a try?

**Comment by feivellau** [comment]
> @feivellau, thanks for your interest! We finetuned the entire model including encoder and decoder. I guess it may not work with efficientsam encoder and original sam decoder. Maybe you can give it a try?

@yformer I have tried it and it doesn't work properly. Furthermore, I have found that the results are very poor when setting `multimask_output=False` to get a single mask.

**Comment by yformer** [comment]
@feivellau, the entire model is finetuned. It may not work properly with efficientsam encoder and original sam decoder. For a single mask, you can set multimask_output=True and then obtain the single mask from output with single_mask = predicted_logits[0, 0, 0, :, :] >= 0.

**Comment by feivellau** [comment]
@yformer.
```python
if multimask_output and self.num_multimask_outputs > 1:
    return masks[:, 1:, :], iou_pred[:, 1:]
else:
    return masks[:, :1, :], iou_pred[:, :1]
```
I'm just trying to follow what the original SAM paper said about a couple different mask outputs. But the results `masks[:, :1, :], iou_pred[:, :1]` in EfficientSAM are not good. And ` masks[:, 1:, :], iou_pred[:, 1:]` needs to be sorted to determine the best output

**Comment by feivellau** [comment]
The same issue as [Expose multimask_output in EfficientSam.forward](https://github.com/yformer/EfficientSAM/issues/30 ) has some examples.
